exoscale/resource_exoscale_nlb_service.go

```diff
@@ -0,0 +1,417 @@
+
+	return resourceNLBServiceRead(d, meta)
+}
+
+func resourceNLBServiceDelete(d *schema.ResourceData, meta interface{}) error {
+	log.Printf("[DEBUG] %s: beginning delete", resourceNLBServiceIDString(d))
+
+	ctx, cancel := context.WithTimeout(context.Background(), d.Timeout(schema.TimeoutDelete))
+	defer cancel()
+
+	client := GetComputeClient(meta)
+
+	zone := d.Get("zone").(string)
+
+	ctx = apiv2.WithEndpoint(ctx, apiv2.NewReqEndpoint(environment(meta), zone))
+	nlb, err := client.GetNetworkLoadBalancer(
+		ctx,
+		zone,
+		d.Get("nlb_id").(string),
+	)
+	if err != nil {
+		return err
+	}
+
+	err = nlb.DeleteService(
+		ctx,
+		&egoscale.NetworkLoadBalancerService{
+			ID: d.Id(),
+		},
+	)
+	if err != nil {
+		return err
+	}
+
+	log.Printf("[DEBUG] %s: delete finished successfully", resourceNLBServiceIDString(d))
+
+	return nil
+}
+
+func resourceNLBServiceApply(ctx context.Context, client *egoscale.Client, d *schema.ResourceData, service *egoscale.NetworkLoadBalancerService) error {
+	if err := d.Set("name", service.Name); err != nil {
+		return err
+	}
+
+	if err := d.Set("description", service.Description); err != nil {
+		return err
+	}
+
+	if err := d.Set("instance_pool_id", service.InstancePoolID); err != nil {
+		return err
+	}
+
+	if err := d.Set("protocol", service.Protocol); err != nil {
+		return err
+	}
+
+	if err := d.Set("port", service.Port); err != nil {
+		return err
+	}
+
+	if err := d.Set("target_port", service.TargetPort); err != nil {
+		return err
+	}
+
+	if err := d.Set("strategy", service.Strategy); err != nil {
+		return err
+	}
+
+	healthcheck := map[string]interface{}{
+		"mode":     service.Healthcheck.Mode,
+		"port":     int(service.Healthcheck.Port),
+		"interval": int(service.Healthcheck.Interval.Seconds()),
+		"timeout":  int(service.Healthcheck.Timeout.Seconds()),
+		"retries":  int(service.Healthcheck.Retries),
+		"uri":      service.Healthcheck.URI,
+	}
+
+	raw := d.Get("healthcheck").(*schema.Set)
+	set := schema.NewSet(raw.F, []interface{}{healthcheck})
+
+	return d.Set("healthcheck", set)
+}
+
+func findNLBService(ctx context.Context, d *schema.ResourceData, meta interface{}) (*egoscale.NetworkLoadBalancerService, error) {
+	client := GetComputeClient(meta)
+
+	zone, okZone := d.GetOk("zone")
+	nlbID, okNLBID := d.GetOk("nlb_id")
+	if okZone && okNLBID {
+		ctx = apiv2.WithEndpoint(ctx, apiv2.NewReqEndpoint(environment(meta), zone.(string)))
+		nlb, err := client.GetNetworkLoadBalancer(ctx, zone.(string), nlbID.(string))
+		if err != nil {
+			return nil, err
+		}
+		for _, s := range nlb.Services {
+			if s.ID == d.Id() {
+				return s, nil
+			}
+		}
+	}
+
+	ctx = apiv2.WithEndpoint(ctx, apiv2.NewReqEndpoint(environment(meta), ""))
+	resp, err := client.RequestWithContext(ctx, egoscale.ListZones{})
+	if err != nil {
+		return nil, err
+	}
+	zones := resp.(*egoscale.ListZonesResponse).Zone
+
+	for _, zone := range zones {
+		nlbs, err := client.ListNetworkLoadBalancers(ctx, zone.Name)
```

Comment:
Same change described for `findNLB()` has to be done here.

Reply:
https://github.com/terraform-providers/terraform-provider-exoscale/pull/52/commits/e17659e16939890c12f2de6a7ccef9bd13ab3a99